### ISSUE TITLE
torch.save does not save python attributes attached to Tensors or Parameters

### ISSUE THREAD

**Issue by thegregyang** [pytorch/pytorch#72129]
### 🐛 Describe the bug

If one attaches python attributes to a Tensor or Parameter, these attributes are not saved by torch.save. This is not expected based on torch's documentation that ["torch.save() and torch.load() use Python’s pickle by default"](https://pytorch.org/docs/stable/notes/serialization.html#saving-loading-tensors).

```python
import torch
xx = torch.zeros(3, 4)
xx.foo = 'bar'
torch.save(xx, '_xx.pt')
torch.load('_xx.pt').foo
```
yields an error
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_19139/1249207437.py in <module>
----> 1 torch.load('_xx.pt').foo

AttributeError: 'Tensor' object has no attribute 'foo'
```
whereas one would expect this returns `'bar'`.

### Versions

Collecting environment information...
PyTorch version: 1.10.2
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.3 LTS (x86_64)
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.27

Python version: 3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-1067-azure-x86_64-with-glibc2.27
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Tesla P100-PCIE-16GB
Nvidia driver version: 495.46
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.2
[pip3] torch==1.10.2
[pip3] torchaudio==0.10.2
[pip3] torchvision==0.11.3
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               11.3.1               h2bc3f7f_2
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2021.3.0           h06a4308_520
[conda] mkl-service               2.4.0            py39h7f8727e_0
[conda] mkl_fft                   1.3.1            py39hd3c417c_0
[conda] mkl_random                1.2.2            py39h51133e4_0
[conda] numpy                     1.21.2           py39h20f2e39_0
[conda] numpy-base                1.21.2           py39h79a1101_0
[conda] pytorch                   1.10.2          py3.9_cuda11.3_cudnn8.2.0_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.10.2               py39_cu113    pytorch
[conda] torchvision               0.11.3               py39_cu113    pytorch

cc @ezyang @gchanan @zou3519 @mruberry

**Comment by albanD** [comment]
Hi,

This was fixed on master. Can you try with a nightly build to confirm?

**Comment by zou3519** [comment]
Did the fix make it to PyTorch 1.11?

**Comment by albanD** [comment]
After double checking this was fixed only for subclasses and the fastpath for plain Tensors does not handle that.
